src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1972,7 +1972,12 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
                     MethodDesc contextMethod = methodFromContext(pResolvedToken.tokenContext);
                     if (contextMethod == MethodBeingCompiled)
                     {
-                        FieldDesc runtimeDeterminedField = (FieldDesc)GetRuntimeDeterminedObjectForToken(ref pResolvedToken);
+                        FieldDesc runtimeDeterminedField;
+#if READYTORUN
+                        runtimeDeterminedField = field;
```

Comment:
This looks like the same fix you had in #6597 where I put a comment: https://github.com/dotnet/corert/pull/6597/files#r235184347

Could you answer the comment? It's not clear to me where we are going to make it so that this doesn't attempt to just set `MyGen<__Canon>.Which` twice.

Reply:
I have added your example to the base CPAOT unit test suite and I used it to debug the four static access flavors. Indeed I found multiple bugs in the process, interestingly enough the above bit remains the same. I think the trick is that, by the time we're about to access the field, we have already secured the right instantiation parameter so that we can make do with just the canonical version. I may be missing some corner case behavior but I'm seeing exactly the same code generated by Crossgen and CPAOT for generic static field access with this change in the couple of tests I looked at.

Reply:
> interestingly enough the above bit remains the same

Is the information we compute here used by RyuJIT after switching things over to `CORINFO_FIELD_STATIC_GENERICS_STATIC_HELPER` at all?

I.e. if you move the endif from line 1984 all the way down to line 2041 (deleting all the #if READYTORUN in between), do things still work?

Reply:
Again, isn't this a cargo cult?

Crossgen only populates `pResult->fieldLookup` for `CORINFO_FIELD_INSTANCE` and `CORINFO_FIELD_STATIC_SHARED_STATIC_HELPER`. It's not populated for `CORINFO_FIELD_STATIC_GENERICS_STATIC_HELPER` and RyuJIT likely doesn't look at it.

Reply:
OK, after fixing various other issues I was finally able to successfully run #if out the part you suggest so I made this change in the 4th commit.